pubsub_nats.go

```diff
@@ -119,14 +119,33 @@ func (p *natsPubSub) Publish(publication *Publication, opts ...PubSubOptPublish)
 
 func (p *natsPubSub) Subscribe(pubs chan *Publication, errors chan error, topic string, opts ...PubSubOptSubscribe) func() {
 
-	config := natsSubscribeConfig{}
+	config := natsSubscribeConfig{
+		replyTimeout: 5 * time.Second,
+	}
+
 	for _, opt := range opts {
 		opt(&config)
 	}
 
 	var sub *nats.Subscription
 	var err error
 
+	responseHandler := func(replyAddr string, replyCh <-chan *Publication) {
+		select {
+		case r := <-replyCh:
+			r.Topic = replyAddr
+			if err := p.Publish(r); err != nil {
+				errors <- err
+			}
+		// TODO: the publisher should be able to provide a response deadline for the publication to the
+		// subscriber so that the subscriber knows when to give up in the event that processing has taken
+		// too long and will no longer be fruitful (e.g. the client that made the request no longer cares
+		// about the reply because you took too long to respond).
+		case <-time.After(config.replyTimeout):
```

Comment:
Why am I not using a context here?

Can't because of how client code uses Subscribe. It's called once with all the options they want (i.e. a context with a deadline) and then they process the messages getting sent to the publication & error channel they created. See the example code in the PR description.

Reply:
I'm not sure I understand the reason

Reply:
A client would typically consume this API like so:

```go
	pubsCh := make(chan *bahamut.Publication)
	errsCh := make(chan error)

	disco := pubsub.Subscribe(pubsCh,
                       errsCh, 
                       "some topic of interest", 
                       bahamut.NATSOptSubscribeQueue("main"))

        // start reading from channels you provided!
	for {
		select {
		case p := <-pubsCh:
			go func(pub *bahamut.Publication) {

				// do some processing here...



				// whenever you are ready, respond to the publication
				response := bahamut.NewPublication("")
				response.Data = []byte("this is my response!")

				pub.Reply(response)

			}(p)
		case err := <-errsCh:
			fmt.Printf("received an error: %+v", err)
		}
	}
```

If I were to configure a context via some option, it would only be useful **once**.